src/ACEtk/block/details/BaseDistributionData/src/generateXSS.hpp

```diff
@@ -0,0 +1,37 @@
+static std::vector< double >
+generateXSS( const std::string& name,
+             std::vector< long >&& boundaries,
+             std::vector< long >&& interpolants,
+             std::vector< Distribution >&& distributions,
+             std::size_t locb ) {
+
+  // add the distribution data to the xss array
+  std::size_t nr = boundaries.size();
+  std::size_t ne = distributions.size();
+  std::vector< double > xss( 1 + 2 * ne );
+  xss[0] = ne;
```

Comment:
Suggest `static_cast<double>(ne)`. Same with other int-to-xss operations elsewhere.